### ISSUE TITLE
What data is used for confusion matrix at the end of training?

### ISSUE THREAD

**Issue by AlexPasqua** [ultralytics/ultralytics#15840]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

What is the data split used to compute the confusion matrix at the end of a training run? Validation, test, ...?

I tried launching short trainings with different version of the dataset: a complete one (train, val and test), one without validation data and one without test data. All of them create the confusion matrix at the end of the training procedure.

**Comment by Y-T-G** [comment]
It uses the `val` set.

**Comment by AlexPasqua** [comment]
> It uses the `val` set.

It seems it works if you don't have a val set, but you have a test set, right?
Also, at this point, what is the test set used for during the training procedure?

**Comment by Y-T-G** [comment]
Okay, yes. If you don't have `val`, it will fallback to `test`. If you have both, then `test` is never used:

https://github.com/ultralytics/ultralytics/blob/848c5cd5062f320dbb2fdcfa1602c6144e23398b/ultralytics/engine/trainer.py#L538

**Comment by AlexPasqua** [comment]
Thank you @Y-T-G 🙂

**Comment by Jordan-Pierce** [comment]
Wow, can't believe that isn't documented on the docs page more clearly; thanks for pointing that out @Y-T-G; does this hold true for classification as well as object detection tasks? 

I guess if you have a test set, you could run a separate evaluation on it after training / evaluation on the validation set?

**Comment by Y-T-G** [comment]
@Jordan-Pierce None of the tasks use the test set during training by default AFAIK. You can use it by explicitly running `model.val(..., split="test")`.

**Comment by Jordan-Pierce** [comment]
@Y-T-G , yep, I just made the change. I'm surprised that when running `.val()` on any of the splits, you can't change the output folder; it's baked in. The only way to get them in the training folder is to pass in `exists_ok`, which just overwrites the previous evaluations?

**Comment by Y-T-G** [comment]
@Jordan-Pierce The val runs are considered distinct from train. You can specify the subfolder with the `name` argument. But it won't merge with existing folder that has the same name.

**Comment by Jordan-Pierce** [comment]
@Y-T-G  sub-folder! Thanks so much for your help!

You wouldn't happen to know how to change the confusion matrix figsize?

**Comment by Y-T-G** [comment]
@Jordan-Pierce The figsize is hardcoded so there's no way of changing it without modifying the code.

You could plot the confusion matrix manually from the returned matrix after `model.val()` `results.confusion_matrix.matrix`

**Comment by Jordan-Pierce** [comment]
Much appreciate @Y-T-G, glad you're on the tickets page!
